packages/repo/src/verify.ts

```diff
@@ -50,6 +41,25 @@ export const verifyCheckout = async (
   }
 }
 
+export const verifyCheckoutToRepo = async (
+  storage: ReadableBlockstore,
+  head: CID,
+  did: string,
+  signingKey: string,
+): Promise<ReadableRepo> => {
+  const repo = await ReadableRepo.load(storage, head)
+  if (repo.did !== did) {
+    throw new RepoVerificationError(`Invalid repo did: ${repo.did}`)
+  }
+  const validSig = await util.verifyCommitSig(repo.commit, signingKey)
+  if (!validSig) {
+    throw new RepoVerificationError(
+      `Invalid signature on commit: ${repo.cid.toString()}`,
+    )
+  }
+  return repo
```

Comment:
one concern here is that without walking the datastore (with a diff), we're not actually verifying that the repo is properly structured

so if you say `verifyCheckoutToRepo` expecting that the repo you get is verified, it may be poorly formatted once you get down into the datastore.

That being said, if you try to do anything with a poorly formatted part, it'll still error so probably not a huge concern 🤔

Reply:
Ah, good point!